### ISSUE TITLE
If you solve the Master Ore quest, quest is not marked as solved

### ISSUE THREAD

**Issue by Wuzzy2** [wrightmat/zbom#105]
If you complete the Master Ore quest, you get a new sword, but the quest is not marked as completed in the quest log.

If you continue to talk to the blacksmith, he still talks about the master ore as if you have never touched it.

Version of ZBOM: 1.4
Version of Solarus: 1.6.0

**Comment by wrightmat** [comment]
Looks like a variable didn't get set somehow. I'm guessing you got the sword when you went to leave the cave? Any errors or anything?

**Comment by Wuzzy2** [comment]
Yes, I got the sword by first giving the master ore to the black smith, then I was leaving the case and the blacksmith stopped be to give me the new sword.

There were no error messages or stuff like that. I just noticed later the quest was not marked as solved later.

**Comment by wrightmat** [comment]
Unable to recreate - likely already patched.
